### PR TITLE
Initial support for Oracle Cloud Infrastructure (OCI) intel module.

### PR DESCRIPTION
**Pull request by Relys**

### CODE REVIEW

**Comment by kunaals** [comment]
@Relys this PR is amazing!! Any chance you can remove the copyrights on each file so that the open source community can legally use this? Happy to have further discussions regarding this matter as well - my email is ksikka@lyft.com.

File: README.md [hunk 1]
```diff
@@ -17,10 +17,12 @@ You can learn more about the story behind Cartography in our [presentation at BS
 Start [here](https://lyft.github.io/cartography/install.html).
 
 ## Supported platforms
+
 - [Amazon Web Services](https://lyft.github.io/cartography/modules/aws/index.html) - API Gateway, Config, EC2, ECS, ECR, Elasticsearch, Elastic Kubernetes Service (EKS), DynamoDB, IAM, KMS, Lambda, RDS, Redshift, Route53, S3, Secrets Manager, Security Hub, SQS, SSM, STS, Tags
 - [Google Cloud Platform](https://lyft.github.io/cartography/modules/gcp/index.html) - Cloud Resource Manager, Compute, DNS, Storage, Google Kubernetes Engine
 - [Google GSuite](https://lyft.github.io/cartography/modules/gsuite/index.html) - users, groups
 - [Duo CRXcavator](https://lyft.github.io/cartography/modules/crxcavator/index.html) - Chrome extensions, GSuite users
+- [Oracle Cloud Infrastructure](docs/setup/config/oci.md) - IAM
```
Comment (achantavy) [inline README.md:25]:
```suggestion
- [Oracle Cloud Infrastructure](https://lyft.github.io/cartography/modules/oci/index.html) - IAM
```

File: cartography/data/jobs/cleanup/oci_import_compartments_cleanup.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "statements": [{
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCICompartment) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_compartments_cleanup.json:3]:
```suggestion
    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCICompartment) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n)",
```

File: cartography/data/jobs/cleanup/oci_import_groups_cleanup.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "statements": [{
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCIGroup) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_groups_cleanup.json:3]:
```suggestion
    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCIGroup) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n)",
```

File: cartography/data/jobs/cleanup/oci_import_groups_cleanup.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "iterative": true,
+    "iterationsize": 100
+  },
+  {
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[r:RESOURCE]->(:OCIGroup) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_groups_cleanup.json:8]:
```suggestion
    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[r:RESOURCE]->(:OCIGroup) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r)",
```

File: cartography/data/jobs/cleanup/oci_import_groups_membership_cleanup.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "statements": [{
+    "query": "MATCH (:OCITenancy{ocidocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(:OCI_User)-[r:MEMBER_OCI_GROUP]->(:OCIGroup) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_groups_membership_cleanup.json:3]:
```suggestion
    "query": "MATCH (:OCITenancy{ocidocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(:OCI_User)-[r:MEMBER_OCI_GROUP]->(:OCIGroup) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r)",
```

File: cartography/data/jobs/cleanup/oci_import_policies_cleanup.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+  "statements": [{
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:OCI_POLICY]->(n:OCIPolicy) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+    "iterative": true,
+    "iterationsize": 100
+  },
+  {
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[r:OCI_POLICY]->(:OCIPolicy) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_policies_cleanup.json:8]:
```suggestion
    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[r:OCI_POLICY]->(:OCIPolicy) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r)",
```

File: cartography/data/jobs/cleanup/oci_import_users_cleanup.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "statements": [{
+    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCIUser) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_import_users_cleanup.json:3]:
```suggestion
    "query": "MATCH (:OCITenancy{ocid: {OCI_TENANCY_ID}})-[:RESOURCE]->(n:OCIUser) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n)",
```

File: cartography/data/jobs/cleanup/oci_tenancy_cleanup.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "statements": [
+    {
+      "query": "MATCH (n:OCITenancy) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/oci_tenancy_cleanup.json:4]:
```suggestion
      "query": "MATCH (n:OCITenancy) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n)",
```

File: cartography/intel/oci/__init__.py [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+        # compute.sync(neo4j_session, resources.compute,
+        #   tenancy_id, region["name"], oci_sync_tag, common_job_parameters
+        # )
+        # network.sync(neo4j_session, resources.network,
+        #   tenancy_id, region["name"], oci_sync_tag, common_job_parameters
+        # )
+
+    # Look into adding once DNS records are implemented.
+    # NOTE clean up all DNS records, regardless of which job created them
+    # run_cleanup_job('OCI_account_dns_cleanup.json', neo4j_session, common_job_parameters)
```
Comment (achantavy) [inline cartography/intel/oci/__init__.py:50]:
```suggestion
```

File: cartography/intel/oci/organizations.py [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+                "configured correctly, your OCI config file is valid, and your credentials have the required Audit "
+                "policies attached (https://docs.cloud.oracle.com/iaas/Content/Identity/Concepts/commonpolicies.htm)."
+            ),
+            e,
+        )
+        return {}
+
+
+def get_oci_profile_names_from_config() -> List[Any]:
+    config_path = oci.config._get_config_path_with_fallback("~/.oci/config")
```
Comment (achantavy) [inline cartography/intel/oci/organizations.py:42]: Putting a pin on this one to make this configurable soon™️

**Review by csanders-git — APPROVED** [review]

**Review by achantavy — APPROVED** [review]
Thank you @Relys for submitting this; we finally merged it in!

Thank you @csanders-git for helping getting this over the line!

**Pull request merged**